### PR TITLE
Add Jib Core and CLI release checklists and update release workflows

### PR DESCRIPTION
**Pull request by chanseokoh**
This auto-files a checklist issue when releasing Jib Core and Jib CLI.

### CODE REVIEW

File: .github/RELEASE_TEMPLATES/cli_release_checklist.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+title: CLI Release {{ env.RELEASE_NAME }}
+labels: release
+---
+## Requirements
+- [ ] ⚠️ Ensure the release process has succeeded before proceeding
+
+## Github
+- [ ] Update [CHANGELOG.md]({{ env.CHANGELOG_URL }})
+- [ ] Update [README.md]({{ env.README_URL }})
+- [ ] Complete [Release]({{ env.RELEASE_DRAFT }})
```
Comment (mpeddada1) [inline .github/RELEASE_TEMPLATES/cli_release_checklist.md:11]:
```suggestion
- [ ] Publish [Release]({{ env.RELEASE_DRAFT }})
- [ ] Merge Release PR ({{ env.RELEASE_PR }})
```

File: .github/RELEASE_TEMPLATES/core_release_checklist.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+title: Core Release {{ env.RELEASE_NAME }}
+labels: release
+---
+## Requirements
+- [ ] ⚠️ Ensure the release process has succeeded before proceeding
+
+## Github
+- [ ] Update [CHANGELOG.md]({{ env.CHANGELOG_URL }})
+- [ ] Update [README.md]({{ env.README_URL }})
+- [ ] Complete [Release]({{ env.RELEASE_DRAFT }})
```
Comment (chanseokoh) [inline .github/RELEASE_TEMPLATES/core_release_checklist.md:11]:
```suggestion
- [ ] Publish [Release]({{ env.RELEASE_DRAFT }})
```

File: .github/RELEASE_TEMPLATES/cli_release_checklist.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+labels: release
+---
+## Requirements
+- [ ] ⚠️ Ensure the release process has succeeded before proceeding
+
+## Github
+- [ ] Update [CHANGELOG.md]({{ env.CHANGELOG_URL }})
+- [ ] Update [README.md]({{ env.README_URL }})
+- [ ] Publish [Release]({{ env.RELEASE_DRAFT }})
+- [ ] Merge Release PR ({{ env.RELEASE_PR }})
```
Comment (mpeddada1) [inline .github/RELEASE_TEMPLATES/cli_release_checklist.md:12]: oh sorry, we also need to add a RELEASE_PR environment variable.

**Review by mpeddada1 — APPROVED** [review]

File: .github/RELEASE_TEMPLATES/cli_release_checklist.md (hunk 3 above)
Comment (chanseokoh) [reply .github/RELEASE_TEMPLATES/cli_release_checklist.md]: Good catch! I mistakenly thought the CLI release process doesn't create a version update PR. (Worse, I removed the checklist item in Core, not CLI.)

**Pull request merged**